terraform/projects/app-ecs-instances/instance-user-data.tpl

```diff
@@ -32,10 +32,15 @@ else
   echo "[$(date '+%H:%M:%S %d-%m-%Y')] attach-volume: /dev/$DEVICE is already formatted: $(file -s /dev/$DEVICE)"
 fi
 
-# Allow prometheus container user access to read/write/execute within container
+# Allow prometheus container user access to read/write metrics to persistent volume mount from within a container
 mkdir -p /ecs/prometheus_data
+
+#Cretaing prometheus & granting access to prometheus data files. We create the group & match it to the gid
+groupadd --system --gid 65534 prometheus
+useradd --uid 65534 --gid 65534 --system --groups prometheus
 mount /dev/$DEVICE /ecs/prometheus_data
-chmod 777 /ecs/prometheus_data
+chown prometheus:prometheus /ecs/prometheus
+chmod -R 760 /ecs/prometheus_data
```

Comment:
I don't understand this change. You've created a `prometheus` user and group, but how does that relate to the prometheus docker container? We never specify a user or group for it to run as.

Reply:
Also, don't you need the execute bit on the directory, in order to `cd` into it? shouldn't this be `770` not `760`?

Reply:
Where do the UID/GID come from? Can you link to some docs/Dockerfile source or something to show why this works?

Reply:
I was wondering about where the UID/GID come from too. I was of the understanding that it runs as `nobody` and therefore is `99/99` as per https://github.com/prometheus/prometheus/issues/3441

Reply:
@idavidmcdonald This is a value that is static within the docker container. We use this value in order to set a predictable UID/GID. 

We have the option to possibly set it manually within the container however this will add unnecessary complexity. I believe this is the best method thus far.